app/Main.hs

```diff
@@ -1,6 +1,85 @@
 module Main where
 
-import Text.Printf (printf)
+import Entities
+import Control.Monad.IO.Class  (liftIO)
+import Database.Persist
+import Database.Persist.Sqlite
+import Text.Megaparsec
+import FooParser (parseSections, Section(..))
+import qualified Database.Esqueleto as E
 
 main :: IO ()
-main = printf "foo"
+main = do
+  fileContents <- readFile "specs.foo"
+
+  -- parse specs.foo
+  let sections = case runParser parseSections "specs.foo" fileContents of
+        Left err        -> error $ "could not parse .foo file with error: " ++ show err
+        Right sections' -> sections'
+
+  -- run everything inside single connection/txn against an in-memory
+  -- instance of SQLite
+  runSqlite ":memory:" $ do
+
+    -- run a safe migration
+    runMigration migrateAll
+
+    -- prepare users, insert users, get entities
+    let users = map (\section -> User $ sectionUsername section) sections :: [User]
+    _ <- insertMany users
```

Comment:
What I was getting at is, won't this crash if you have users with duplicate names? Do you want to handle that in some more graceful way?